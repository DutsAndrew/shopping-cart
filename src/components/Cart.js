import React, { useState, useEffect } from "react";
import '../style/Cart.css';

const Cart = props => {

  const { cart, incrementGame, decrementGame, removeGame } = props;

  useEffect(() => {
    console.log(cart);
  }, [cart]);

  return (
    <div id="cart-container">
      <div id="cart-title-container">
        <p id="cart-title">Cart</p>
        <ul id="cart-game-tiles">
          {cart.cart.map((game) => {
            return <div className="cart-game-card" key={game.id}>
              <img className='cart-game-card-img' src={game.img} alt={game.title} ></img>
                <p className='cart-game-card-title' >{game.title}</p>
                <p className='cart-game-card-price' >{game.price}</p>
                <p className="cart-game-card-quantity">Quantity: {game.quantity}</p>
                <div id="increment-buttons-container">
                  <button id="increment-item" onClick={incrementGame} >Add Item</button>
                  <button id="decrement-item" onClick={decrementGame} >Remove Item</button>
                </div>
                <button id="remove-from-cart-button" onClick={removeGame} >Remove from Cart</button>
            </div>
          })}
        </ul>
      </div>
      <div id="cart-total-container">
        <p id="total-amount-label-text">Your Total:</p>
        <p id="total-amount-text">${Math.abs(cart.total.toFixed(2))}</p>
        <button id="checkout-button">Proceed to Checkout</button>
      </div>
    </div>
  );
};

export default Cart;