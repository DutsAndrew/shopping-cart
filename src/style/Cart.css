#cart-container {
  position: absolute;
  display: flex;
  flex-flow: row wrap;
  justify-content: flex-start;
  align-items: center;
  flex: auto;
  right: 28%;
  margin-top: 6%;
  width: 70vw;
  height: auto;
  background-color: #c7d5e0;
  color: #2a475e;
}

#cart-title-container {
  display: flex;
  justify-content: center;
}

#cart-title {
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 2em;
  height: 1vh;
  width: 70vw;
}

#cart-total-container{
  display: flex;
  flex-flow: column nowrap;
  justify-content: flex-start;
  align-items: center;
  position: absolute;
  left: 101%;
  top: 0%;
  width: 25vw;
  height: 40vh;
  background-color: #c7d5e0;
  color: #2a475e;
}

#total-amount-label-text {
  margin-top: 25%;
  font-size: 1.5em;
}

#total-amount-text {
  font-size: 1.25em;
}

#checkout-button {
  margin-top: 15%;
  width: 20vw;
  height: 3vh;
  font-size: 1em;
  border-radius: 4%;
  background-color: #2a475e;
  color: #c7d5e0;
}

#cart-game-tiles {
  position: absolute;
  display: flex;
  flex-flow: row wrap;
  align-items: center;
  justify-content: center;
  align-content: center;
  flex: auto;
  margin: 0;
  margin-top: 4%;
  top: 80%;
  right: 0.1%;
  width: 70vw;
  height: auto;
  gap: 2%;
}

.cart-game-card {
  display: flex;
  flex-flow: column nowrap;
  justify-content: center;
  align-items: center;
  align-content: center;
  margin-bottom: 2%;
  width: 22vw;
  height: 40vh;
  background-color: rgb(27, 40, 56, 1);
  color: #c7d5e0;
}

.cart-game-card-img {
  width: 15vw;
  height: 10vh;
}

.cart-game-card-title {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 20vw;
  font-size: 1.1em;
}

.cart-game-card-price {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 20vw;
  font-size: 0.95em;
}

#increment-buttons-container {
  display: flex;
  flex-flow: row nowrap;
  justify-content: center;
  align-items: center;
  gap: 2%;
  width: 20vw;
  height: 2vh;
}

#increment-item, #decrement-item {
  background-color: #c7d5e0;
  color: #2a475e;
}

#remove-from-cart-button {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 15vw;
  height: 2vh;
  margin-top: 2%;
  background-color: #c7d5e0;
  color: #2a475e;
}