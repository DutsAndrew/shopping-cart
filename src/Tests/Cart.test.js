import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Cart from '../components/Cart';

describe('Cart component', () => {
  it('renders Cart', () => {
    const cart = {
      cart: [],
      quantity: 0,
      total: 0,
    }
    const incrementGame = jest.fn();
    const decrementGame = jest.fn();
    const removeGame = jest.fn();
    const { container } = render(<Cart cart={cart}
      incrementGame={incrementGame}
      decrementGame={decrementGame}
      removeGame={removeGame}
      />);
    expect(container).toMatchSnapshot();
  });
});