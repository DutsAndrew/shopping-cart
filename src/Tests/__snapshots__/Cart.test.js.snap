// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart component renders Cart 1`] = `
<div>
  <div
    id="cart-container"
  >
    <div
      id="cart-title-container"
    >
      <p
        id="cart-title"
      >
        Cart
      </p>
      <ul
        id="cart-game-tiles"
      />
    </div>
    <div
      id="cart-total-container"
    >
      <p
        id="total-amount-label-text"
      >
        Your Total:
      </p>
      <p
        id="total-amount-text"
      >
        $
        0
      </p>
      <button
        id="checkout-button"
      >
        Proceed to Checkout
      </button>
    </div>
  </div>
</div>
`;
